src/liballoc/borrow.rs

```diff
@@ -479,3 +490,18 @@ impl<'a> AddAssign<Cow<'a, str>> for Cow<'a, str> {
         }
     }
 }
+
+#[stable(feature = "cow_str_add_assign_char", since = "1.41.0")]
+impl AddAssign<char> for Cow<'_, str> {
+    fn add_assign(&mut self, rhs: char) {
+        if let Cow::Borrowed(lhs) = *self {
+            let base_capacity = lhs.len() + rhs.len_utf8();
+            //Attempt amortized memory allocation
```

Comment:
Is it possible to avoid duplicating this sizing heuristic from the existing amortized_new_size in raw_vec.rs? I thought you had used amortized_new_size successfully in an earlier revision.

If we make changes to the heuristic in the future, it should only be in one place.